Code in :plugin-annotations is not resolvable in JS box test while JVM's one is ok

I try to use annotation in box test, i try jvm box test first, it goes well. And then i try the js box test, it throw out exceptions below:

```markdown
Starting Gradle Daemon...
Gradle Daemon started in 627 ms
Reusing configuration cache.
> Task :compiler-plugin:checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compiler-plugin:generateTestBuildConfig SKIPPED
> Task :compiler-plugin:generateTestFixturesBuildConfig SKIPPED
> Task :compiler-plugin:processTestFixturesResources NO-SOURCE
> Task :plugin-annotations:jvmProcessResources NO-SOURCE
> Task :plugin-annotations:processJvmMainResources SKIPPED
> Task :plugin-annotations:kmpPartiallyResolvedDependenciesChecker
> Task :plugin-annotations:checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compiler-plugin:processResources UP-TO-DATE
> Task :compiler-plugin:processTestResources UP-TO-DATE
> Task :compiler-plugin:kotlinWasmD8Setup UP-TO-DATE
> Task :compiler-plugin:generateBuildConfig UP-TO-DATE
> Task :plugin-annotations:compileKotlinJvm UP-TO-DATE
> Task :plugin-annotations:compileJvmMainJava NO-SOURCE
> Task :plugin-annotations:jvmMainClasses UP-TO-DATE
> Task :compiler-plugin:compileKotlin UP-TO-DATE
> Task :compiler-plugin:compileJava NO-SOURCE
> Task :compiler-plugin:classes UP-TO-DATE
> Task :plugin-annotations:jvmJar UP-TO-DATE
> Task :compiler-plugin:jar UP-TO-DATE
> Task :compiler-plugin:compileTestFixturesKotlin UP-TO-DATE
> Task :compiler-plugin:compileTestFixturesJava NO-SOURCE
> Task :compiler-plugin:testFixturesClasses UP-TO-DATE
> Task :compiler-plugin:testFixturesJar UP-TO-DATE
> Task :compiler-plugin:generateTests UP-TO-DATE
> Task :compiler-plugin:compileTestKotlin NO-SOURCE
> Task :compiler-plugin:compileTestJava UP-TO-DATE
> Task :compiler-plugin:testClasses UP-TO-DATE
info: KLIB loader: Not a Kotlin library, or a library with invalid format: /Users/miku/Downloads/compiler-plugin-template-master/plugin-annotations/build/libs/plugin-annotations-jvm-0.1.0-SNAPSHOT.jar

Multiple Failures (3 failures)
	org.opentest4j.AssertionFailedError: Actual data differs from file content: simple.kt
	org.opentest4j.AssertionFailedError: Content is not equal: simple.fir.txt
	java.lang.IllegalStateException: Serialization of IrErrorType is not supported anymore
org.opentest4j.MultipleFailuresError: Multiple Failures (3 failures)
	org.opentest4j.AssertionFailedError: Actual data differs from file content: simple.kt
	org.opentest4j.AssertionFailedError: Content is not equal: simple.fir.txt
	java.lang.IllegalStateException: Serialization of IrErrorType is not supported anymore
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:80)
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:54)
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:48)
	at org.junit.jupiter.api.Assertions.assertAll(Assertions.java:2978)
	at org.jetbrains.kotlin.test.services.JUnit5Assertions.failAll(JUnit5Assertions.kt:90)
	at org.jetbrains.kotlin.test.TestRunner.reportFailures(TestRunner.kt:125)
	at org.jetbrains.kotlin.test.TestRunner.runTestPipeline(TestRunner.kt:117)
	at org.jetbrains.kotlin.test.TestRunner.runTestImpl(TestRunner.kt:70)
	at org.jetbrains.kotlin.test.TestRunner.runTest(TestRunner.kt:30)
	at org.jetbrains.kotlin.test.TestRunner.runTest$default(TestRunner.kt:28)
	at org.jetbrains.kotlin.test.runners.AbstractKotlinCompilerTest.runTest(AbstractKotlinCompilerTest.kt:115)
	at org.jetbrains.kotlin.compiler.plugin.template.runners.JsBoxTestGenerated.testSimple(JsBoxTestGenerated.java:32)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
	Suppressed: org.opentest4j.AssertionFailedError: Actual data differs from file content: simple.kt
		at org.jetbrains.kotlin.test.services.JUnit5Assertions.assertEqualsToFile(JUnit5Assertions.kt:64)
		at org.jetbrains.kotlin.test.Assertions.assertEqualsToFile(Assertions.kt:15)
		at org.jetbrains.kotlin.test.Assertions.assertEqualsToFile$default(Assertions.kt:14)
		at org.jetbrains.kotlin.test.services.GlobalMetadataInfoHandler.compareAllMetaDataInfos(GlobalMetadataInfoHandler.kt:81)
		at org.jetbrains.kotlin.test.TestRunner.runTestPipeline(TestRunner.kt:107)
		... 12 more
	Suppressed: org.opentest4j.AssertionFailedError: Content is not equal: simple.fir.txt
		at org.jetbrains.kotlin.test.services.JUnit5Assertions.assertEqualsToFile(JUnit5Assertions.kt:64)
		at org.jetbrains.kotlin.test.Assertions.assertEqualsToFile$default(Assertions.kt:24)
		at org.jetbrains.kotlin.test.frontend.fir.handlers.FirDumpHandler.processAfterAllModules(FirDumpHandler.kt:87)
		at org.jetbrains.kotlin.test.TestRunner.runTestPipeline(TestRunner.kt:101)
		... 12 more
	Suppressed: java.lang.IllegalStateException: Serialization of IrErrorType is not supported anymore
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.getToIrTypeDeduplicationKey(IrFileSerializer.kt:455)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrType(IrFileSerializer.kt:474)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeNameAndType(IrFileSerializer.kt:1161)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrVariable(IrFileSerializer.kt:1318)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeDeclaration(IrFileSerializer.kt:1412)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeStatement(IrFileSerializer.kt:1121)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeBlockBody(IrFileSerializer.kt:483)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeStatement(IrFileSerializer.kt:1127)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrStatementBody(IrFileSerializer.kt:213)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrFunctionBase(IrFileSerializer.kt:1223)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrFunction(IrFileSerializer.kt:1237)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeDeclaration(IrFileSerializer.kt:1408)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeTopLevelDeclaration(IrFileSerializer.kt:1672)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrFile$lambda$0(IrFileSerializer.kt:1594)
		at org.jetbrains.kotlin.backend.common.serialization.signature.PublicIdSignatureComputer.inFile(IdSignatureComputers.kt:43)
		at org.jetbrains.kotlin.backend.common.serialization.DeclarationTable.inFile(DeclarationTable.kt:65)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.inFile(IrFileSerializer.kt:1570)
		at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrFile(IrFileSerializer.kt:1576)
		at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeIrFile(IrModuleSerializer.kt:36)
		at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializedIrModule(IrModuleSerializer.kt:47)
		at org.jetbrains.kotlin.backend.common.serialization.SerializeModuleIntoKlibKt.serializeModuleIntoKlib(serializeModuleIntoKlib.kt:122)
		at org.jetbrains.kotlin.ir.backend.js.KlibKt.serializeModuleIntoKlib(klib.kt:512)
		at org.jetbrains.kotlin.cli.pipeline.web.WebKlibSerializationPipelinePhase.executePhase(WebKlibSerializationPipelinePhase.kt:42)
		at org.jetbrains.kotlin.js.test.converters.FirKlibSerializerCliWebFacade.transform(FirKlibSerializerCliWebFacade.kt:51)
		at org.jetbrains.kotlin.js.test.converters.FirKlibSerializerCliWebFacade.transform(FirKlibSerializerCliWebFacade.kt:28)
		at org.jetbrains.kotlin.test.TestStep$FacadeStep.processModule(TestStep.kt:47)
		at org.jetbrains.kotlin.test.TestRunner.processModule(TestRunner.kt:206)
		at org.jetbrains.kotlin.test.TestRunner.hackyProcessModule(TestRunner.kt:197)
		at org.jetbrains.kotlin.test.TestRunner.processModule(TestRunner.kt:141)
		at org.jetbrains.kotlin.test.TestRunner.runTestPipeline(TestRunner.kt:92)
		... 12 more


Actual data differs from file content: simple.kt
<点击以查看差异>

org.opentest4j.AssertionFailedError: Actual data differs from file content: simple.kt
	at org.jetbrains.kotlin.test.services.JUnit5Assertions.assertEqualsToFile(JUnit5Assertions.kt:64)
	at org.jetbrains.kotlin.test.Assertions.assertEqualsToFile(Assertions.kt:15)
	at org.jetbrains.kotlin.test.Assertions.assertEqualsToFile$default(Assertions.kt:14)
	at org.jetbrains.kotlin.test.services.GlobalMetadataInfoHandler.compareAllMetaDataInfos(GlobalMetadataInfoHandler.kt:81)
	at org.jetbrains.kotlin.test.TestRunner.runTestPipeline(TestRunner.kt:107)
	at org.jetbrains.kotlin.test.TestRunner.runTestImpl(TestRunner.kt:70)
	at org.jetbrains.kotlin.test.TestRunner.runTest(TestRunner.kt:30)
	at org.jetbrains.kotlin.test.TestRunner.runTest$default(TestRunner.kt:28)
	at org.jetbrains.kotlin.test.runners.AbstractKotlinCompilerTest.runTest(AbstractKotlinCompilerTest.kt:115)
	at org.jetbrains.kotlin.compiler.plugin.template.runners.JsBoxTestGenerated.testSimple(JsBoxTestGenerated.java:32)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)


Content is not equal: simple.fir.txt
<点击以查看差异>

org.opentest4j.AssertionFailedError: Content is not equal: simple.fir.txt
	at org.jetbrains.kotlin.test.services.JUnit5Assertions.assertEqualsToFile(JUnit5Assertions.kt:64)
	at org.jetbrains.kotlin.test.Assertions.assertEqualsToFile$default(Assertions.kt:24)
	at org.jetbrains.kotlin.test.frontend.fir.handlers.FirDumpHandler.processAfterAllModules(FirDumpHandler.kt:87)
	at org.jetbrains.kotlin.test.TestRunner.runTestPipeline(TestRunner.kt:101)
	at org.jetbrains.kotlin.test.TestRunner.runTestImpl(TestRunner.kt:70)
	at org.jetbrains.kotlin.test.TestRunner.runTest(TestRunner.kt:30)
	at org.jetbrains.kotlin.test.TestRunner.runTest$default(TestRunner.kt:28)
	at org.jetbrains.kotlin.test.runners.AbstractKotlinCompilerTest.runTest(AbstractKotlinCompilerTest.kt:115)
	at org.jetbrains.kotlin.compiler.plugin.template.runners.JsBoxTestGenerated.testSimple(JsBoxTestGenerated.java:32)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)


Serialization of IrErrorType is not supported anymore
java.lang.IllegalStateException: Serialization of IrErrorType is not supported anymore
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.getToIrTypeDeduplicationKey(IrFileSerializer.kt:455)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrType(IrFileSerializer.kt:474)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeNameAndType(IrFileSerializer.kt:1161)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrVariable(IrFileSerializer.kt:1318)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeDeclaration(IrFileSerializer.kt:1412)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeStatement(IrFileSerializer.kt:1121)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeBlockBody(IrFileSerializer.kt:483)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeStatement(IrFileSerializer.kt:1127)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrStatementBody(IrFileSerializer.kt:213)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrFunctionBase(IrFileSerializer.kt:1223)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrFunction(IrFileSerializer.kt:1237)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeDeclaration(IrFileSerializer.kt:1408)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeTopLevelDeclaration(IrFileSerializer.kt:1672)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrFile$lambda$0(IrFileSerializer.kt:1594)
	at org.jetbrains.kotlin.backend.common.serialization.signature.PublicIdSignatureComputer.inFile(IdSignatureComputers.kt:43)
	at org.jetbrains.kotlin.backend.common.serialization.DeclarationTable.inFile(DeclarationTable.kt:65)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.inFile(IrFileSerializer.kt:1570)
	at org.jetbrains.kotlin.backend.common.serialization.IrFileSerializer.serializeIrFile(IrFileSerializer.kt:1576)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializeIrFile(IrModuleSerializer.kt:36)
	at org.jetbrains.kotlin.backend.common.serialization.IrModuleSerializer.serializedIrModule(IrModuleSerializer.kt:47)
	at org.jetbrains.kotlin.backend.common.serialization.SerializeModuleIntoKlibKt.serializeModuleIntoKlib(serializeModuleIntoKlib.kt:122)
	at org.jetbrains.kotlin.ir.backend.js.KlibKt.serializeModuleIntoKlib(klib.kt:512)
	at org.jetbrains.kotlin.cli.pipeline.web.WebKlibSerializationPipelinePhase.executePhase(WebKlibSerializationPipelinePhase.kt:42)
	at org.jetbrains.kotlin.js.test.converters.FirKlibSerializerCliWebFacade.transform(FirKlibSerializerCliWebFacade.kt:51)
	at org.jetbrains.kotlin.js.test.converters.FirKlibSerializerCliWebFacade.transform(FirKlibSerializerCliWebFacade.kt:28)
	at org.jetbrains.kotlin.test.TestStep$FacadeStep.processModule(TestStep.kt:47)
	at org.jetbrains.kotlin.test.TestRunner.processModule(TestRunner.kt:206)
	at org.jetbrains.kotlin.test.TestRunner.hackyProcessModule(TestRunner.kt:197)
	at org.jetbrains.kotlin.test.TestRunner.processModule(TestRunner.kt:141)
	at org.jetbrains.kotlin.test.TestRunner.runTestPipeline(TestRunner.kt:92)
	at org.jetbrains.kotlin.test.TestRunner.runTestImpl(TestRunner.kt:70)
	at org.jetbrains.kotlin.test.TestRunner.runTest(TestRunner.kt:30)
	at org.jetbrains.kotlin.test.TestRunner.runTest$default(TestRunner.kt:28)
	at org.jetbrains.kotlin.test.runners.AbstractKotlinCompilerTest.runTest(AbstractKotlinCompilerTest.kt:115)
	at org.jetbrains.kotlin.compiler.plugin.template.runners.JsBoxTestGenerated.testSimple(JsBoxTestGenerated.java:32)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)


> Task :compiler-plugin:test FAILED
JsBoxTestGenerated > testSimple() FAILED
    org.gradle.internal.exceptions.DefaultMultiCauseException at JsBoxTestGenerated.java:32
        Caused by: org.opentest4j.AssertionFailedError at JsBoxTestGenerated.java:32
1 test completed, 1 failed
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':compiler-plugin:test'.
> There were failing tests. See the report at: file:///Users/miku/Downloads/compiler-plugin-template-master/compiler-plugin/build/reports/tests/test/index.html
* Try:
> Run with --scan to get full insights.
BUILD FAILED in 3s
14 actionable tasks: 2 executed, 12 up-to-date
Configuration cache entry reused.


```

simple.kt
```markdown
package foo.bar

import org.jetbrains.kotlin.compiler.plugin.template.SomeAnnotation
import org.jetbrains.kotlin.compiler.plugin.template.testApp

@SomeAnnotation
fun box(): String {
    val result = MyClass().foo()
    val app = testApp()
    return if (result == "Hello world") {
        "OK"
    } else {
        "Fail: $result"
    }
}
```


simple.kt (result)
```markdown
package foo.bar

import org.<!UNRESOLVED_IMPORT!>jetbrains<!>.kotlin.compiler.plugin.template.SomeAnnotation
import org.<!UNRESOLVED_IMPORT!>jetbrains<!>.kotlin.compiler.plugin.template.testApp

@<!UNRESOLVED_REFERENCE!>SomeAnnotation<!>
fun box(): String {
    val result = MyClass().foo()
    val app = <!UNRESOLVED_REFERENCE!>testApp<!>()
    return if (result == "Hello world") {
        "OK"
    } else {
        "Fail: $result"
    }
}

```

--- a/compiler-plugin/build.gradle.kts
+++ b/compiler-plugin/build.gradle.kts
@@ -1,6 +1,8 @@
 @file:OptIn(ExperimentalWasmDsl::class)
 
 import org.jetbrains.kotlin.gradle.ExperimentalWasmDsl
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinUsages
 import org.jetbrains.kotlin.gradle.targets.wasm.d8.D8EnvSpec
 import org.jetbrains.kotlin.gradle.targets.wasm.d8.D8Plugin
 
@@ -32,8 +34,21 @@ idea {
     module.generatedSourceDirs.add(projectDir.resolve("test-gen"))
 }
 
-val annotationsRuntimeClasspath: Configuration by configurations.creating { isTransitive = false }
 val testArtifacts: Configuration by configurations.creating
+
+val annotationsRuntimeClasspath by configurations.dependencyScope("annotationsRuntimeClasspath") {
+    isTransitive = false
+}
+val annotationsJvmRuntimeClasspath by configurations.resolvable("annotationsJvmRuntimeClasspath") {
+    extendsFrom(annotationsRuntimeClasspath)
+}
+val annotationsJsRuntimeClasspath by configurations.resolvable("annotationsJsRuntimeClasspath") {
+    extendsFrom(annotationsRuntimeClasspath)
+    attributes {
+        attribute(Usage.USAGE_ATTRIBUTE, objects.named(KotlinUsages.KOTLIN_RUNTIME))
+        attribute(KotlinPlatformType.attribute, KotlinPlatformType.js)
+    }
+}
 
 dependencies {
     compileOnly(libs.kotlin.compiler)
@@ -67,12 +82,15 @@ buildConfig {
 }
 
 tasks.test {
-    dependsOn(annotationsRuntimeClasspath)
+    dependsOn(testArtifacts)
+    dependsOn(annotationsJvmRuntimeClasspath)
+    dependsOn(annotationsJsRuntimeClasspath)
 
     useJUnitPlatform()
     workingDir = rootDir
 
-    systemProperty("annotationsRuntime.classpath", annotationsRuntimeClasspath.asPath)
+    systemProperty("annotationsRuntime.jvm.classpath", annotationsJvmRuntimeClasspath.asPath)
+    systemProperty("annotationsRuntime.js.classpath", annotationsJsRuntimeClasspath.asPath)
 
     // Properties required to run the internal test framework.
     setLibraryProperty("org.jetbrains.kotlin.test.kotlin-stdlib", "kotlin-stdlib")

--- a/compiler-plugin/test-fixtures/org/jetbrains/kotlin/compiler/plugin/template/services/PluginAnnotationsProvider.kt
+++ b/compiler-plugin/test-fixtures/org/jetbrains/kotlin/compiler/plugin/template/services/PluginAnnotationsProvider.kt
@@ -2,16 +2,16 @@ package org.jetbrains.kotlin.compiler.plugin.template.services
 
 import org.jetbrains.kotlin.cli.jvm.config.addJvmClasspathRoots
 import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.js.config.JSConfigurationKeys
+import org.jetbrains.kotlin.platform.isJs
+import org.jetbrains.kotlin.platform.jvm.isJvm
 import org.jetbrains.kotlin.test.builders.TestConfigurationBuilder
 import org.jetbrains.kotlin.test.model.TestModule
 import org.jetbrains.kotlin.test.services.EnvironmentConfigurator
 import org.jetbrains.kotlin.test.services.RuntimeClasspathProvider
 import org.jetbrains.kotlin.test.services.TestServices
+import org.jetbrains.kotlin.test.services.targetPlatform
 import java.io.File
-
-private val annotationsRuntimeClasspath =
-    System.getProperty("annotationsRuntime.classpath")?.split(File.pathSeparator)?.map(::File)
-        ?: error("Unable to get a valid classpath from 'annotationsRuntime.classpath' property")
 
 fun TestConfigurationBuilder.configureAnnotations() {
     useConfigurators(::PluginAnnotationsProvider)
@@ -20,10 +20,39 @@ fun TestConfigurationBuilder.configureAnnotations() {
 
 private class PluginAnnotationsProvider(testServices: TestServices) : EnvironmentConfigurator(testServices) {
     override fun configureCompilerConfiguration(configuration: CompilerConfiguration, module: TestModule) {
-        configuration.addJvmClasspathRoots(annotationsRuntimeClasspath)
+        val platform = module.targetPlatform(testServices)
+        when {
+            platform.isJvm() -> {
+                configuration.addJvmClasspathRoots(annotationsJvmRuntimeClasspath)
+            }
+
+            platform.isJs() -> {
+                val libraries = configuration.getList(JSConfigurationKeys.LIBRARIES)
+                configuration.put(
+                    JSConfigurationKeys.LIBRARIES,
+                    libraries + annotationsJsRuntimeClasspath.map { it.absolutePath },
+                )
+            }
+        }
     }
 }
 
 private class PluginAnnotationsClasspathProvider(testServices: TestServices) : RuntimeClasspathProvider(testServices) {
-    override fun runtimeClassPaths(module: TestModule) = annotationsRuntimeClasspath
+    override fun runtimeClassPaths(module: TestModule): List<File> {
+        val targetPlatform = module.targetPlatform(testServices)
+        return when {
+            targetPlatform.isJvm() -> annotationsJvmRuntimeClasspath
+            targetPlatform.isJs() -> annotationsJsRuntimeClasspath
+            else -> emptyList()
+        }
+    }
+}
+
+private val annotationsJvmRuntimeClasspath = classpathFiles("annotationsRuntime.jvm.classpath")
+private val annotationsJsRuntimeClasspath = classpathFiles("annotationsRuntime.js.classpath")
+
+private fun classpathFiles(property: String): List<File> {
+    val property = System.getProperty(property)
+        ?: error("Unable to get a valid classpath from '$property' property")
+    return property.split(File.pathSeparator).map(::File)
 }

--- a/compiler-plugin/testData/box/simple.fir.ir.txt
+++ b/compiler-plugin/testData/box/simple.fir.ir.txt
@@ -1,5 +1,7 @@
 FILE fqName:foo.bar fileName:/simple.kt
   FUN name:box visibility:public modality:FINAL returnType:kotlin.String
+    annotations:
+      SomeAnnotation
     BLOCK_BODY
       VAR name:result type:kotlin.String [val]
         CALL 'public final fun foo (): kotlin.String declared in foo.bar.MyClass' type=kotlin.String origin=null

--- a/compiler-plugin/testData/box/simple.fir.txt
+++ b/compiler-plugin/testData/box/simple.fir.txt
@@ -1,7 +1,7 @@
 FILE: simple.kt
     package foo.bar
 
-    public final fun box(): R|kotlin/String| {
+    @R|org/jetbrains/kotlin/compiler/plugin/template/SomeAnnotation|() public final fun box(): R|kotlin/String| {
         lval result: R|kotlin/String| = R|foo/bar/MyClass|().R|foo/bar/MyClass.foo|()
         ^box when () {
             ==(R|<local>/result|, String(Hello world)) ->  {

--- a/compiler-plugin/testData/box/simple.kt
+++ b/compiler-plugin/testData/box/simple.kt
@@ -1,5 +1,8 @@
 package foo.bar
 
+import org.jetbrains.kotlin.compiler.plugin.template.SomeAnnotation
+
+@SomeAnnotation
 fun box(): String {
     val result = MyClass().foo()
     return if (result == "Hello world") { "OK" } else { "Fail: $result" }

